src/App.tsx

```diff
@@ -1,24 +1,49 @@
-import React, { useState, useCallback } from 'react';
+import React, { useState, useCallback, useEffect } from 'react';
 import axios from 'axios';
 import styled from 'styled-components';
 import PhotosTable from './PhotosTable';
 import PhotoMetaData from './PhotoMetaData';
 
 import './App.css';
-import { Photo } from './types';
+import { Photo, Facets, GraphState, GraphType, GRAPH_TYPES } from './types';
 import Navbar from './layout/Navbar';
 import FileUplaoder from './FileUploader';
+import GraphContainer from './GraphContainer';
 
 const App: React.FC = () => {
   const [photos, updatePhotos] = useState<Photo[]>([]);
   const [currentPhoto, updateCurrentPhoto] = useState<Photo | null>(null);
+  const [currentFacets, updateFacets] = useState<Facets>({});
```

Comment:
for bigger state objects you can use `useReducer` hook it works +/- the same way as redux reducers